src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const RESULT = [];
+  let newObj = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        newObj = Object.assign(newObj, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newObj[key];
+        }
+        break;
+
+      case 'clear':
+        newObj = {};
+    }
```

Comment:
[CODE STYLE]: The `switch` statement should include a `default` case to handle any unexpected `action.type` values. This is for error handling and to ensure that the function behaves predictably even with incorrect input.